cibuildwheel/action.yml

```diff
@@ -0,0 +1,169 @@
+          cibw_before_all="${{ inputs.cibw-before-all-amd64 }}"
+          auditwheel_repair_command="${{ inputs.auditwheel-repair-command-amd64 }}"
+        elif [ "${arch}" == "aarch64" ]; then
+          cibw_before_all="${{ inputs.cibw-before-all-arm64 }}"
+          auditwheel_repair_command="${{ inputs.auditwheel-repair-command-arm64 }}"
+        fi
+        cibw_before_all="$cibw_before_all_ssh $cibw_before_all"
+
+        # set before-all-linux command to:
+        #   1. set up ssh-agent + deploy key for cloning private repos
+        #   2. store internal pypi credentials in .netrc
+        #   3. append `&& cibw_before_all` if it's not empty
+        if [ "${cibw_before_all}" != "" ]; then
+          echo "CIBW_BEFORE_ALL_LINUX=printf 'machine pypi.k8s.rapids.ai\n\tlogin ${{ inputs.internal-pypi-user }}\n\tpassword ${{ inputs.internal-pypi-pass }}\n' > ~/.netrc && ${cibw_before_all}" >> "${GITHUB_ENV}"
+        else
+          echo "CIBW_BEFORE_ALL_LINUX=printf 'machine pypi.k8s.rapids.ai\n\tlogin ${{ inputs.internal-pypi-user }}\n\tpassword ${{ inputs.internal-pypi-pass }}\n' > ~/.netrc" >> "${GITHUB_ENV}"
+        fi
+
+        # set repair command based on runner arch
+        echo "CIBW_REPAIR_WHEEL_COMMAND_LINUX=${auditwheel_repair_command}" >> "${GITHUB_ENV}"
+
+        # set CIBW_BUILD based on input python version e.g. 3.8 -> cp38-*
+        py_vers="${{ inputs.python-version }}"
+        cibw_build_str=""
+        for pyver in $py_vers; do
+          cibw_build_str="${cibw_build_str}cp${pyver//./}-* "
+        done
+        echo "CIBW_BUILD=$cibw_build_str" >> "${GITHUB_ENV}"
+
+        # enable sccache in skbuild config options
+        skbuild_config_opts="${{ inputs.skbuild-configure-options}}"
+        skbuild_config_opts="${skbuild_config_opts} -DCMAKE_C_COMPILER_LAUNCHER=/usr/bin/sccache -DCMAKE_CXX_COMPILER_LAUNCHER=/usr/bin/sccache -DCMAKE_CUDA_COMPILER_LAUNCHER=/usr/bin/sccache"
+
+        # assemble CIBW_ENVIRONMENT from skbuild configure and build options
+        # StrictHostKeyChecking=no is to avoid (yes/no) prompt from private repo clone
+        echo "CIBW_ENVIRONMENT=GIT_SSH_COMMAND='ssh -i /private-deploy-key -o StrictHostKeyChecking=no' SKBUILD_CONFIGURE_OPTIONS='${skbuild_config_opts}' SKBUILD_BUILD_OPTIONS='${{ inputs.skbuild-build-options }}' ${{ inputs.cibw-environment }} RAPIDS_PY_WHEEL_CUDA_SUFFIX='${{ inputs.python-package-cuda-suffix }}' PYTHON_PACKAGE_CUDA_SUFFIX='${{ inputs.python-package-cuda-suffix }}' RAPIDS_PY_WHEEL_VERSIONEER_OVERRIDE='${{ inputs.python-package-versioneer-override }}' SCCACHE_S3_KEY_PREFIX=gha-cibw SCCACHE_REGION=us-west-2 SCCACHE_IDLE_TIMEOUT=32768 SCCACHE_BUCKET=rapids-sccache AWS_ACCESS_KEY_ID=${{ inputs.sccache-aws-access-key-id }} AWS_SECRET_ACCESS_KEY=${{ inputs.sccache-aws-secret-access-key }} PIP_INDEX_URL=https://pypi.k8s.rapids.ai/simple" >> "${GITHUB_ENV}"
+      shell: bash
+
+    - name: Invoke cibuildwheel with pipx
+      run: >
+        pipx run --verbose
```

Comment:
See if we really need pipx, or just install cibuildwheel straight up in the 'thin launcher container'.

Reply:
Enforce `cibuildwheel==2.8.0` in the thin launcher container. Upgrading cibuildwheel = rebuilding the launcher container.